### DISCUSSION TITLE
Session and Test Isolation Problem

### DISCUSSION THREAD

**Discussion by sevdavc** [cypress-io/cypress#25415] [Questions and Help] [answered]
I have a problem with test isolation. I enabled test isolation and I want to use session for my login process:
```
cy.session([email, password], () => {
    cy.visit("/login")
    cy.get('[data-test="login-username"]').type(email)
    cy.get('[data-test="login-password"]').type(password)
    cy.get('[data-test="login-button"]').click()
  },{validate: () => {
    cy.hash().should('eq', '/dashboard')
  }})
cy.visit('/dashboard')
```

This is my session, if there's anything wrong please let me know. Then I run my tests and there's no problem when it's creating the session but if its restoring the session,default blank page appears. How can i fix it? 

2- I have one more question. I override some of my tests with {testIsolation: false}, and can't i use session for that part too in before section? If I can, please enlighten me. 
```
describe('Folder', { testIsolation: false }, () => {
  before(() => {
    cy.loginWithoutSession('admin', 'admin')
    cy.get('[data-test="navbar-pages-option"]').should('be.visible').click()
  })
```

**Comment by emilyrohrbough** [comment, marked as answer]
1. > if its restoring the session, default blank page appears. How can i fix it?

The page is blank due to test isolation. The page is navigated to `about:blank` before each test so when your session is restored, your validation logic will always fail when verifying the route with `cy.hash().should('eq', '/dashboard')`. I suggested moving this check into the setup and using the validate func to verify the client is still authenticated with the backend.

2. Yes you can use cy.session when test isolation is disabled.

**Reply by sevdavc** [reply]
Thanks for helping! 
```
cy.session([email, password], () => {
    cy.visit("/login")
    cy.get('[data-test="login-username"]').type(email)
    cy.get('[data-test="login-password"]').type(password)
    cy.get('[data-test="login-button"]').click()
    cy.hash().should('eq', '#/dashboard')
  },{
    validate: () => {
    expect(localStorage.getItem('token')).to.be.not.null
  }})
```
This session worked for me.
